orderer/common/multichannel/registrar.go

```diff
@@ -1044,3 +1066,18 @@ func (r *Registrar) removeSystemChannel() error {
 
 	return nil
 }
+
+func (r *Registrar) removeLedgerAsync(channelID string) {
+	r.pendingRemoval[channelID] = true
+
+	go func() {
+		if err := r.ledgerFactory.Remove(channelID); err != nil {
+			r.pendingRemoval[channelID] = false
```

Comment:
this needs to enveloped in the lock unlock as well; 
delete() is better, `delete(r.pendingRemoval, channelID)` to avoid keeping all those removed channels in the map.

Reply:
As you noted later, it's better to keep these failed removals in the map to indicate failure on further attempts to join or remove the same channel.